dbt/include/sqlserver/macros/materializations/test.sql

```diff
@@ -0,0 +1,12 @@
+{% macro sqlserver__get_test_sql(main_sql, fail_calc, warn_if, error_if, limit) -%}
+    select
+      {{ fail_calc }} as failures,
+      case when {{ fail_calc }} {{ warn_if }}
+        then 1 else 0 end as should_warn,
+      case when {{ fail_calc }} {{ error_if }}
+        then 1 else 0 end as should_error
```

Comment:
```suggestion
      case when {{ fail_calc }} {{ warn_if }}
        then true else false end as should_warn,
      case when {{ fail_calc }} {{ error_if }}
        then true else false end as should_error
```
dbt [expects these](https://github.com/fishtown-analytics/dbt/blob/c794600242e734d870d4bf1c8fb4c0349ab961eb/core/dbt/task/test.py#L36-L37) to be boolean. This may work by type coercion, I'm not sure

Reply:
cool cool cool. but what if our favorite "byzantine" (h/t  @jeroen-mostert lol) dialect doesn't support a boolean type? all it has is `BIT`... Lemme try the local dev mode install and get back to you with what I find out

https://stackoverflow.com/a/32150794/3842610

Reply:
Ah... in that case, 0/1 may work, "true"/"false" (strings) may work, it just needs to be close enough to validate [here](https://github.com/fishtown-analytics/dbt/blob/c794600242e734d870d4bf1c8fb4c0349ab961eb/core/dbt/task/test.py#L108)